activesupport/lib/active_support/core_ext/string/output_safety.rb

```diff
@@ -289,10 +289,32 @@ def #{unsafe_method}!(*args, &block)            # def gsub!(*args, &block)
     end
 
     private
-      def html_escape_interpolated_argument(arg)
+      def explicit_html_escape_interpolated_argument(arg)
         (!html_safe? || arg.html_safe?) ? arg : CGI.escapeHTML(arg.to_s)
       end
 
+      def implicit_html_escape_interpolated_argument(arg)
+        if !html_safe? || arg.html_safe?
+          arg
+        else
+          arg_string = begin
+            arg.to_str
+          rescue NoMethodError => error
+            if error.name == :to_str
+              str = arg.to_s
+              ActiveSupport::Deprecation.warn <<~MSG.squish
+                Implicit conversion of #{arg.class} into String by ActiveSupport::SafeBuffer is deprecated.
+                You must explictly cast it to a String.
```

Comment:
I'm not quite up to date on the deprecations policies. Should this message say it will be removed in 7.1 or 7.2?

Reply:
Deprecations are pretty much always the version after so 7.1

Reply:
Thanks. Added that target.